### DISCUSSION TITLE
Disable Focus update

### DISCUSSION THREAD

**Discussion by patexoid** [lllyasviel/Fooocus#1382] [Q&A] [answered]
Looks like after latest update it started to generate more arms, legs or eyes  than I expect, and other images for the same configuration and seed comparing to previous version.
Is it possible to disable fooocus autoupdate, and do it manually when required? 
Or even roll back to previous version?

**Comment by lllyasviel** [comment]
hi you can use git checkout to use any version in https://github.com/lllyasviel/Fooocus/commits/main

to avoid update, you can disconnect internet when using fooocus or manually launch the program with `python launch.py`

if you find anything going wrong with any version, please report an issue if necessary.

**Comment by rlovell3** [comment]
Agreed.  I've seen a major degradation of photo styled shots from yesterday to today, using same prompts, seeds, ....  Identical settings.  I'm rolling back.

**Reply by rlovell3** [reply]
Just checked version numbers.  Good was 2.1.825.  Current one giving me concerns: 2.1.835 and also 836

**Comment by lllyasviel** [comment, marked as answer]
Hi, we noticed the difference between current version and 2.1.824.

The difference is caused by some numeric float point difference in attention computation. The result is not better or worse but is slightly different in some random seed. 

We are not adding or removing anything to the Fooocus algorithm. The effect is similar to updating Nvidia driver when Nvidia occasionally changes how numbers are multiplied.

You can use archived 2.1.824 here https://github.com/lllyasviel/Fooocus/tree/v2.1.824 if you are in a professional studio that requires 100% same results as before. To use it, the commands are 

    cd Fooocus
    git pull
    git switch v2.1.824
    python launch.py

By using this, all updating will be turned off forever.

**Reply by rlovell3** [reply]
Your stuff is too good to turn off updates, but thanks for the info.  I'm willing to follow the commit changes to see where I might want to deviate.  Speaking of deviations, I've written my own args_manager.py to force the issue on a few of your optional flags.  Otherwise, enjoying the ride, and not in a position to complain about the great work! Keep going !!!!

**Reply by lllyasviel** [reply]
Update:

also added 

    v2.1.786 

and 

    v2.1.774

These two versions also have numeric float point difference between each other and may be helpful if people want to try.

**Reply by lllyasviel** [reply]
Update:

also added

    v2.0.780

**Reply by R4290N** [reply]
I apologize, please.
I am very thankful to you for your wonderful product. and I like the latest update very much, but apparently I will have to use 2.1.824 along with the latest version for some time to come.

but I don't understand how to revert back to 2.1.824 using the commands and links you mentioned. please give more detailed instructions

**Reply by lllyasviel** [reply]
Hi everyone, we now have a document for previous versions. See here:
https://github.com/lllyasviel/Fooocus/discussions/1405

**Comment by patexoid** [comment]
thanks for help

**Comment by lllyasviel** [comment]
Hi everyone, we now have a document for previous versions. See here:
https://github.com/lllyasviel/Fooocus/discussions/1405
